front/src/modules/pathfinding/components/IncompatibleConstraints/utils.ts

```diff
@@ -0,0 +1,64 @@
+import { Zap, Gauge, Signals } from '@osrd-project/ui-icons';
+import { featureCollection, type FeatureCollection } from '@turf/helpers';
+import lineSliceAlong from '@turf/line-slice-along';
+import type { LineString } from 'geojson';
+import { sortBy, uniq } from 'lodash';
+
+import type { IncompatibleConstraintItemEnhanced } from './type';
+
+export function getIcon(type: string) {
+  switch (type) {
+    case 'incompatible_electrification_ranges':
+      return Zap;
+    case 'incompatible_gauge_ranges':
+      return Gauge;
+    case 'incompatible_signalisation_system_ranges':
+      return Signals;
+    default:
+      return Zap;
```

Comment:
How about a `?` here?

Reply:
I returned a default one because we need one (linter rule), and normally we should never use it. 
A `?` is not an Icon component